src/dialogs/AdvancedCriteriaRowValue.js

```diff
@@ -0,0 +1,176 @@
+    });
+  };
+
+  const removeFilter = () => {
+    const newArray = [...filters];
+    newArray.splice(index, 1);
+    setFilters(newArray.length === 0 ? [CLEARED_STATE_FILTER] : newArray);
+  };
+
+  const renderInputBasedOnType = (type) => {
+    const commonProps = {
+      module: "paymentPlan",
+      label: "paymentPlan.advancedCriteria.value",
+      value: currentFilter.value,
+      onChange: onAttributeChange("value"),
+    };
+  
+    switch (type) {
+      case BOOLEAN:
+        return (
+          <SelectInput
+            options={BOOL_OPTIONS}
+            {...commonProps}
+          />
+        );
+      case INTEGER:
+        return (
+          <NumberInput
+            min={0}
+            displayZero
+            {...commonProps}
+          />
+        );
+      case STRING:
+      default:
+        if (currentFilter.field.toLowerCase().includes(DATE)) {
+          return (
+            <PublishedComponent
+              pubRef="core.DatePicker"
+              {...commonProps}
+            />
+          );
+        } else {
+          return (
+            <TextInput
+              {...commonProps}
+            />
+          );
+        }
+    }
+  };
+
+  return (
+    <Grid 
+      container 
+      direction="row" 
+      className={classes.item}
+      style={{ backgroundColor: "#DFEDEF" }}
+    >
+      {filters.length > 0 ? (
+        <div style={{ backgroundColor: '#DFEDEF', width: '10px', height: '25px', marginTop: '25px' }}>
+          <span
+            style={{
+              transform: 'translate(-50%, -50%)',
+              fontSize: '16px',
+              color: '#006273',
+            }}
+            onClick={removeFilter}
+          >
+            &#x2716;
+          </span>
+        </div> 
+      ) : (<></>)
+      }
+      <Grid item xs={3} className={classes.item}>
+        <CustomFilterFieldStatusPicker
+          module="paymentPlan"
+          label="paymentPlan.advancedCriteria.field"
+          value={{ field: currentFilter.field, type: currentFilter.type }}
+          onChange={onAttributeChange("field")}
+          customFilters={customFilters}
+        />
+      </Grid>
+        {currentFilter.field !== "" ? (
+          <Grid item xs={3} className={classes.item}>
+            <CustomFilterTypeStatusPicker
+              module="paymentPlan"
+              label="paymentPlan.advancedCriteria.filter"
+              value={currentFilter.filter}
+              onChange={onAttributeChange("filter")}
+              customFilters={customFilters}
+              customFilterField={currentFilter.field}
+            />
+          </Grid>
+        ) : (<></>) }
+        {currentFilter.field !== "" && currentFilter.filter !== "" ? (
+          <Grid item xs={3} className={classes.item}>
+            {renderInputBasedOnType(currentFilter.type)}
+          </Grid>
+        ) : (<></>) }
+        {currentFilter.field !== "" && currentFilter.filter !== "" && currentFilter.value !== "" ? (
```

Comment:
I'm wondering if we should consider creating a constant for the empty string. Throughout this PR, the empty string has been used multiple times, and addressing changes in each instance might prove time-consuming. I'd like to hear your thoughts on this idea.

Reply:
I think we can skip this. This is how the functionality works on modal level and would be hard to change.